chia/cmds/farm_funcs.py

```diff
@@ -211,23 +212,40 @@ async def summary(rpc_port: int, wallet_rpc_port: int, harvester_rpc_port: int,
         print(f"Block rewards: {(amounts['farmer_reward_amount'] + amounts['pool_reward_amount']) / units['chia']}")
         print(f"Last height farmed: {amounts['last_height_farmed']}")
 
-    total_plot_size = 0
-    total_plots = 0
-    if all_plots is not None:
-        for harvester_ip, plots in all_plots.items():
-            if harvester_ip == "success":
-                # This key is just "success": True
-                continue
-            total_plot_size_harvester = sum(map(lambda x: x["file_size"], plots["plots"]))
-            total_plot_size += total_plot_size_harvester
-            total_plots += len(plots["plots"])
-            print(f"Harvester {harvester_ip}:")
-            print(f"   {len(plots['plots'])} plots of size: {format_bytes(total_plot_size_harvester)}")
-
-        print(f"Plot count for all harvesters: {total_plots}")
+    class PlotStats:
+        total_plot_size = 0
+        total_plots = 0
+
+    if all_harvesters is not None:
+        harvesters_local: dict = {}
+        harvesters_remote: dict = {}
+        for harvester in all_harvesters["harvesters"]:
+            ip = harvester["connection"]["host"]
+            if is_localhost(ip):
+                harvesters_local[harvester["connection"]["node_id"]] = harvester
+            else:
+                if ip not in harvesters_remote:
+                    harvesters_remote[ip] = {}
+                harvesters_remote[ip][harvester["connection"]["node_id"]] = harvester
+
+        def print_harvesters(harvester_peers_in: dict):
+            for harvester_peer_id, plots in harvester_peers_in.items():
+                total_plot_size_harvester = sum(map(lambda x: x["file_size"], plots["plots"]))
+                PlotStats.total_plot_size += total_plot_size_harvester
+                PlotStats.total_plots += len(plots["plots"])
+                print(f"   {len(plots['plots'])} plots of size: {format_bytes(total_plot_size_harvester)}")
+
+        if len(harvesters_local):
```

Comment:
i prefer if len(x) > 0 instead of this, i think it's clearer

Reply:
Agree, see b390211198328fbc73265a5cdf46988d36e4c058